influxdb_client/client/write/dataframe_serializer.py

```diff
@@ -5,115 +5,155 @@
-            fields_indexes.append(index_value)
-        elif issubclass(value.type, (np.float, np.bool_)):
-            fields.append(f"{key_format}={{p[{index_value}]}}")
-            fields_indexes.append(index_value)
+            if null_columns[index]:
+                key_value = f"""{{
+                    '' if {val_format} == '' or type({val_format}) == float and math.isnan({val_format}) else
+                    f',{key_format}={{str({val_format}).translate(_ESCAPE_STRING)}}'
+                }}"""
+            else:
+                key_value = f',{key_format}={{str({val_format}).translate(_ESCAPE_KEY)}}'
+            tags.append(key_value)
+            continue
+        # Note: no comma separator needed for the first field.
+        # It's important to omit it because when the first
+        # field column has no nulls, we don't run the comma-removal
+        # regexp substitution step.
+        sep = '' if len(field_indexes) == 0 else ','
+        if issubclass(value.type, np.integer):
+            field_value = f"{sep}{key_format}={{{val_format}}}i"
+        elif issubclass(value.type, np.bool_):
+            field_value = f'{sep}{key_format}={{{val_format}}}'
+        elif issubclass(value.type, np.float):
+            if null_columns[index]:
+                field_value = f"""{{"" if math.isnan({val_format}) else f"{sep}{key_format}={{{val_format}}}"}}"""
+            else:
+                field_value = f'{sep}{key_format}={{{val_format}}}'
         else:
-            fields.append(f"{key_format}=\"{{str(p[{index_value}]).translate(_ESCAPE_STRING)}}\"")
-            fields_indexes.append(index_value)
-
-    tags.sort(key=lambda x: x['key'])
-    tags = ','.join(map(lambda y: y['value'], tags))
-
-    fmt = ('{measurement_name}', f'{"," if tags else ""}', tags,
-           ' ', ','.join(fields), ' {p[0].value}')
-    f = eval("lambda p: f'{}'".format(''.join(fmt)),
-             {'measurement_name': measurement_name, '_ESCAPE_KEY': _ESCAPE_KEY, '_ESCAPE_STRING': _ESCAPE_STRING,
-              'keys': keys})
+            if null_columns[index]:
+                field_value = f"""{{
+                    '' if type({val_format}) == float64 and math.isnan({val_format}) else
+                    f'{sep}{key_format}="{{str({val_format}).translate(_ESCAPE_STRING)}}"'
+                }}"""
+            else:
+                field_value = f'''{sep}{key_format}="{{str({val_format}).translate(_ESCAPE_STRING)}}"'''
+        field_indexes.append(field_index)
+        fields.append(field_value)
+
+    measurement_name = str(data_frame_measurement_name).translate(_ESCAPE_MEASUREMENT)
+
+    tags = ''.join(tags)
+    fields = ''.join(fields)
+    timestamp = '{p[0].value}'
+
+    print(f'measurement_name: {measurement_name}')
+    print(f'keys: {keys}')
+    print(f'tag columns: {data_frame_tag_columns}')
+    print(f'lambda p: f"""{{measurement_name}}{tags} {fields} {timestamp}"""')
```

Comment:
Are the `print`s meant to stay?

Reply:
Nope, that's just me debugging. This PR is still a draft as yet.